src/NLog/LogFactory.cs

```diff
@@ -1175,9 +1175,32 @@ public LogFactory LoadConfiguration(string configFile)
         public LogFactory ReloadConfiguration()
         {
             InternalLogger.Info("Reloading configuration...");
-            Configuration = Configuration?.ReloadNewConfig();
+            var errorRaised = false;
 
-            NotifyConfigurationReloaded();
+            if (_config == null)
+            {
+                try
+                {
+                    Configuration = Configuration.ReloadNewConfig();
```

Comment:
I was actually thinking the other way. When _config == null then you just need to load Configuration. When _config != null then you should call `_config.ReloadNewConfig()`.

Reply:
Done but same as below, so resolve this one